vinyldns/resource_record_set.go

```diff
@@ -104,8 +105,48 @@ func resourceVinylDNSRecordSetRead(d *schema.ResourceData, meta interface{}) err
 	if err != nil {
 		return err
 	}
+	recordType := strings.ToLower(rs.Type)
+
+	if recordType == "soa" {
+		return errors.New(recordType + " records are not currently supported by vinyldns")
+	}
 
 	d.Set("name", rs.Name)
+	d.Set("zone_id", rs.ZoneID)
+	d.Set("ttl", rs.TTL)
+	d.Set("type", rs.Type)
+	d.Set("owner_group_id", rs.OwnerGroupID)
+
+	if recordType == "cname" {
+		d.Set("record_cname", rs.Records[0].CName)
+
+		return nil
+	}
+
+	if recordType == "txt" {
+		d.Set("record_text", rs.Records[0].Text)
+
+		return nil
+	}
+
+	if recordType == "ns" {
+		recs := make([]interface{}, 0, len(rs.Records))
+
+		for _, r := range rs.Records {
+			recs = append(recs, r.NSDName)
+		}
+
+		d.Set("record_nsdnames", schema.NewSet(schema.HashString, recs))
```

Comment:
@mdb is it necessary to[ check the return](https://www.terraform.io/docs/extend/best-practices/detecting-drift.html#error-checking-aggregate-types) of `d.Set()` here because `schema.NewSet()` returns an aggregate type?

Reply:
@rhenning Thanks - good catch! I pushed a fix for this as well as the same problem pertaining to the `record_addresses`, which you also pointed out.